client/src/components/useraccess/Forms/PasswordResetModal.jsx

```diff
@@ -0,0 +1,124 @@
+  /**
+   * 
+   * 
+   * @param { object } event 
+   * @memberof PasswordResetModal
+   * @returns { object } respose from api call 
+   */
+  handleSubmit(event) {
+    // validate input
+    this.setState({ error: true, disableBtn: true});
+    if(this.state.email === "") {
+    message.innerHTML = `Your email is needed to reset your password`;
+    message.className="red-text center-align";  
+    } else {
+      
+      //Make API call
+      this.props.sendEmail({email: this.state.email})
+      .then(() => {
+        this.setState({ error: false, disableBtn: true});
+        message.innerHTML = `A password reset link has been sent to ${this.state.email}`;
+        message.className="green-text center-align";
+      })
+      .catch((error) => {
+        if (error.response.status === 501) {
+          message.innerHTML ="Ooops! Something went wrong, the server could not process your request at this time. Please try again.";
+          message.className="red-text center-align";
+          this.setState({ error: false, disableBtn: false});
+        } else {
+          message.innerHTML = error.response.data.message;
+          message.className="red-text center-align";
+          this.setState({ error: false, disableBtn: false});
+        }
+      })
+    }
+    // prevent button default action
+    event.preventDefault();
+  }
+
+  cancelPassReset() {
+    this.setState({email: '', loader: false, disableBtn: false
+  });
+    message.innerHTML = 'A password reset link will be sent to this email';
+    message.className="black-text center-align"
+  }
+
+  render() {
+    return(
+    <div className="modal passwordModal" id="passwordResetModal">
+      <div className="container">
+        <div className="col s12 m6 l4 offset-l8">
+          <div className="modal-content">
+            <h5 className="center-align">Password reset</h5>
+              <p className="center-align" id="message">
+              A password reset link will be sent to this email</p>
+              {
+                this.state.error 
+                ? 
+                  <div className="row center-align">
+                    <Loader/>
+                  </div>
+                :
+                null
+              }
+            <form className="col s6 offset-s3" onSubmit={this.handleSubmit}>
+              <div className="input-field">
+                <label htmlFor="email">Email</label>
+                <input type="email" className="validate" value={this.state.email} onChange={this.handleInput}/>
+              </div>
+
+              <div className="input-field">
+                <button className="btn waves-effect waves-ripple green" disabled= {this.state.disableBtn}>Reset</button>
+                <div className="btn waves-effect modal-close modal-action" onClick={this.cancelPassReset}>Cancel</div>
+              </div>
+            </form>
+          </div>   
+        </div>
+      </div>
+    </div>
+    );
+  }
+
+}
+
+function dispatchToProps(dispatch){
```

Comment:
ok lemme get something straight, you map dispatched to props but you didn't mapStateToProps that's why you had to lots of setState in your handleSubmit. Since you're using redux I think it would have been better if you use it fully in this component also. It will reduce all the logic handling in your handleSubmit method